restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -390,6 +391,122 @@ private Object convertToBackendBooleanValue(boolean value, boolean numericBoolea
+
+    DocumentDB db = dbFactory.getDocDataStoreForToken(authToken, headers);
+    JsonSurfer surfer = JsonSurferGson.INSTANCE;
+
+    db = maybeCreateTableAndIndexes(dbFactory, db, keyspace, collection, headers, authToken);
+    List<String> idsWritten = new ArrayList<>();
+    try (BufferedReader reader = new BufferedReader(new InputStreamReader(payload, "UTF-8"))) {
+      Iterator<String> iter = reader.lines().iterator();
+      ExecutionContext context = ExecutionContext.NOOP_CONTEXT;
+      String docsPath = convertToJsonPtr(idPath.get());
+      int chunkIndex = 0;
+      final int CHUNK_SIZE = 250;
+      while (iter.hasNext()) {
+        chunkIndex++;
+        Map<String, String> docsInChunk = new LinkedHashMap<>();
+        while (docsInChunk.size() < CHUNK_SIZE && iter.hasNext()) {
+          String doc = iter.next();
+          JsonNode json = mapper.readTree(doc);
+
+          String docId = UUID.randomUUID().toString();
+          if (idPath.isPresent()) {
+            if (!json.at(docsPath).isTextual()) {
+              throw new ErrorCodeRuntimeException(
+                  ErrorCode.DOCS_API_WRITE_BATCH_INVALID_ID_PATH,
+                  String.format(
+                      "Json Document %s requires a String value at the path %s, found %s."
+                          + " Batch %d failed, %d writes were successful. Repeated requests are "
+                          + "idempotent if the same `idPath` is defined.",
+                      doc,
+                      idPath.get(),
+                      json.at(docsPath).toString(),
+                      chunkIndex,
+                      (chunkIndex - 1) * CHUNK_SIZE));
+            }
+            docId = json.requiredAt(docsPath).asText();
+          }
+          docsInChunk.put(docId, doc);
+        }
+
+        // Write the chunk of (at most) 250 documents by firing a single batch with the row inserts
+        List<Object[]> bindVariableList = new ArrayList<>();
+        DocumentDB finalDb = db;
+        List<String> ids =
+            docsInChunk.entrySet().stream()
+                .map(
+                    data -> {
+                      bindVariableList.addAll(
+                          shredJson(
+                                  surfer,
+                                  finalDb,
+                                  Collections.emptyList(),
+                                  data.getKey(),
+                                  data.getValue(),
+                                  false)
+                              .left);
+                      return data.getKey();
+                    })
+                .collect(Collectors.toList());
+
+        long now = timeSource.currentTimeMicros();
+        try {
+          db.deleteManyThenInsertBatch(
+              keyspace,
+              collection,
+              ids,
+              bindVariableList,
+              Collections.emptyList(),
+              now,
+              context.nested("ASYNC INSERT"));
```

Comment:
Could you add a profiling unit test for this, please?

Reply:
added [here](https://github.com/stargate/stargate/pull/1043/files#diff-93b9952f26f7639ef23892552f8d4059d445215cdf6bef3ed4eaacc93357a399R865)